gridstatus/isone_api/isone_api.py

```diff
@@ -0,0 +1,439 @@
+            pd.DataFrame: Processed DataFrame.
+        """
+        df["BeginDate"] = pd.to_datetime(df["BeginDate"])
+        df["Load"] = pd.to_numeric(df["Load"], errors="coerce")
+        df["LocId"] = pd.to_numeric(df["LocId"], errors="coerce")
+        df = df.rename(columns={"BeginDate": "Interval Start"})
+        return df[["Interval Start", "Location", "LocId", "Load"]]
+
+    def get_realtime_hourly_demand_current_static(self) -> pd.DataFrame:
+        """
+        Get the most recent real-time hourly demand data for default locations.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the real-time hourly demand data for default locations.
+        """
+        url = f"{BASE_URL}/realtimehourlydemand/current"
+        response = self.make_api_call(url)
+
+        if (
+            "HourlyRtDemands" not in response
+            or "HourlyRtDemand" not in response["HourlyRtDemands"]
+        ):
+            raise NoDataFoundException("No real-time hourly demand data found.")
+
+        formatted_data = [
+            {
+                "BeginDate": entry["BeginDate"],
+                "Location": entry["Location"]["$"],
+                "LocId": entry["Location"]["@LocId"],
+                "Load": entry["Load"],
+            }
+            for entry in response["HourlyRtDemands"]["HourlyRtDemand"]
+        ]
+
+        df = pd.DataFrame(formatted_data)
+        return self._handle_demand(df)
+
+    def get_realtime_hourly_demand_current(
+        self,
+        locations: list[str] = None,
+    ) -> pd.DataFrame:
+        """
+        Get the most recent real-time hourly demand data for specified locations.
+
+        Args:
+            locations (list[str], optional): List of specific location names to request data for.
+                                             If None, data for all locations will be retrieved.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the real-time hourly demand data for all requested locations.
+        """
+        if not locations:
+            locations = [
+                loc
+                for loc in ZONE_LOCATIONID_MAP.keys()
+                if loc not in EXCLUDE_FROM_REALTIME_HOURLY_DEMAND
+            ]
+
+        all_data = []
+
+        for location in locations:
+            location_id = ZONE_LOCATIONID_MAP.get(location)
+            if not location_id:
+                log.warning(f"{location}: Not a known ISO NE Hub or Zone for this data")
+                continue
+
+            url = f"{BASE_URL}/realtimehourlydemand/current/location/{location_id}"
+            response = self.make_api_call(url)
+
+            data = response["HourlyRtDemand"]
+            data["Location"] = location
+            data["LocId"] = location_id
+            all_data.append(data)
+
+        df = pd.DataFrame(all_data)
+        return self._handle_demand(df)
+
+    @support_date_range(frequency="D")
+    def get_realtime_hourly_demand_historical_range(
```

Comment:
Instead of separate methods for `current` and `historical_range` data, I think we should have one method, `get_realtime_hourly_demand`. If `date == "latest"` then we use the current endpoint, otherwise the historical endpoint. That matches what we do in our other ISOs.

Reply:
I see what you mean. The `current` methods only return a single hourly interval though, rather than a full day. I was thinking we would probably refrain from using them entirely, since it's tricky to time cron jobs or upsert or otherwise not miss intervals if a downstream system fails. 

I could definitely see removing the `current` methods entirely (but they might be useful). But heard about having a `latest` functionality that uses the `day/{date}/` endpoints, I can work on that.

Reply:
> The current methods only return a single hourly interval

Multiple of our methods only return a single interval when you pass `date="latest"`. For example, `MISO().get_fuel_mix("latest")` returns a single interval.